.github/workflows/ci.yaml

```diff
@@ -30,6 +30,12 @@ jobs:
         # Support for Python 3.13 depends on https://github.com/pytorch/pytorch/issues/130249
         version: ["3.10", "3.11", "3.12"]
         os: [ubuntu-24.04]
+        pytorch_requirements: ["pytorch-cpu-requirements.txt"]
+
+        include:
+          - version: "3.11"
+            os: linux-mi300-1gpu-ossci-iree-org
```

Comment:
This isn't an operating system, so we could rename to "runs-on", similar to here: https://github.com/iree-org/iree/blob/e43c555623122ad3be55673e8caa042dd1505c0c/.github/workflows/pkgci_test_onnx.yml#L28-L38

We could also set a matrix name that is more condensed in the UI: https://github.com/iree-org/iree-turbine/actions/runs/15475532636/job/43569829445?pr=903 (note how the full name gets truncated on the left)
![image](https://github.com/user-attachments/assets/0a7cf02b-288d-4237-9514-5f478bfeb60c)

See https://github.com/iree-org/iree/blob/e43c555623122ad3be55673e8caa042dd1505c0c/.github/workflows/pkgci_test_onnx.yml#L22 up a bit in that file for how to set a name